app/helpers/Intersect.js

```diff
@@ -308,11 +334,20 @@ async function asyncCheckIntersect() {
         });
 
         // intersect the users location with the locations from the authority
-        let tempDayBin = intersectSetIntoBins(
+        let { bins, newIntersections } = intersectSetIntoBins(
           locationArray,
           normalizeAndSortLocations(responseJson.concern_points),
+          intersections,
         );
 
+        // Update intersection store if a user has crossed paths with a new infected case
+        if (newIntersections > 0) {
+          await SetStoreData(INTERSECTIONS_DATA, intersections);
+          newIntersectionsCount += newIntersections;
+        }
+
+        let tempDayBin = bins;
+
         // Update each day's bin with the result from the intersection.  To keep the
         //  difference between no data (==-1) and exposure data (>=0), there
         //  are a total of 3 cases to consider.
```

Comment:
This wasnt your code but can you extract this sum method below on line 354 into a separate method?

There's a lot to process here and it would read better if its something like `dayBins = sumBins(dayBins, bins)`